tests/test_logger.py

```diff
@@ -0,0 +1,149 @@
+"""Unit tests for logger module."""
+
+import logging
+import unittest
+from unittest import mock
+
+from PyQt5.QtCore import QObject
+from PyQt5.QtWidgets import QApplication
+
+from iquip.apps import logger
+
+class _SignallerTest(unittest.TestCase):
+    """Unit tests for the _Signaller class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_signal_connection(self):
+        with mock.patch('iquip.apps.logger._Signaller.signal') as mocked_signal:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            mocked_signal.connect.assert_called_once_with(app.addLog)
+
+
+class LoggingHandlerTest(unittest.TestCase):
+    """Unit tests for the LoggingHandler class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_handler_connection(self):
+        with mock.patch('iquip.apps.logger._Signaller.signal') as mocked_signal:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            mocked_signal.connect.assert_called_once_with(app.addLog)
+
+    def test_emit(self):
+        with mock.patch('iquip.apps.logger.LoggingHandler.emit') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            test_logger = logger.logger
+            app.setLevel("INFO")
+            test_logger.info("hello")
+            mocked_method.assert_called_once()
```

Comment:
I think you should test the called parameters, too.

Reply:
I want to do that but it is little bit difficult because the input to the emit is log record, not a simple text.
It contains several information like time, logger generated by `logging`, I have no idea how to check parameter with `called_with`.

Reply:
You can use [`call_args`](https://docs.python.org/ko/3/library/unittest.mock.html#unittest.mock.Mock.call_args), or similar one.

Reply:
@Aijuh